building-applications/pipelines.md

```diff
@@ -27,10 +25,87 @@ pipeline:
+         <td>Pipeline agents configuration.</td>
+      </tr>
+   </tbody>
+</table>
+
+
+### Pipeline agents configuration
+
+Inside the `pipeline` property, you must specify a list of agents. 
+
+Each agent can be configure with the following properties.
+
+<table><thead><tr><th width="163.33333333333331">Name</th><th width="171">Type</th><th>Description</th></tr></thead><tbody><tr><td>name</td><td>String (required)</td><td></td></tr><tr><td>id</td><td>String (required)</td><td></td></tr><tr><td>type</td><td>String (required)</td><td>The type name of processing to be run. See <a href="../pipeline-agents/ai-actions/">AI Actions</a> for supported types.</td></tr><tr><td>input</td><td><br></td><td>Reference to the topic name</td></tr><tr><td>output</td><td><br></td><td>Reference to the topic name</td></tr><tr><td>configuration</td><td>object</td><td>Given the chosen type, these are the config values used. Refer to the configuration area of each type for more info.</td></tr></tbody></table>
+
+### Agent resources
+
+When deploying a pipeline, some agents might require additional CPU or memory to run properly. Depending on the traffic load and the agent architecture, you might need to scale the pipeline vertically or horizontally, or both.
+
+The `resources` property allows you to specify the `size` (scale vertically) and the `parallelism` (scale horizontally) for the pipeline.
+
+
+```yaml
+topics:
+  ...
+
+resources:
+  parallelism: 3
+  size: 2
+  
+pipeline:
+  ...
+```
 
-<table><thead><tr><th width="114.33333333333331">Root</th><th width="163">Node</th><th>Description</th><th data-hidden>Type</th></tr></thead><tbody><tr><td>module</td><td></td><td></td><td><br></td></tr><tr><td>id</td><td></td><td>Referenceable value of the pipeline</td><td><br></td></tr><tr><td>name</td><td></td><td>The name of the application</td><td><br></td></tr><tr><td>topics</td><td></td><td>A collection of topics that will be created within the tenant, and used to transport data between steps.</td><td>object[]</td></tr><tr><td><br></td><td>name</td><td>The name of the topic. This value must comply with the given messaging platform’s rules.</td><td>String (required)</td></tr><tr><td><br></td><td>creation-mode</td><td><p>Whether the pipeline should attempt to create the topic if it doesn’t exist. Supported values are:</p><ul><li>“create-if-not-exists”</li><li>“None”</li></ul><p>The default value is “none”</p></td><td>String (optional)</td></tr></tbody></table>
+Setting `parallelism:` to 3 the pipeline will be deployed with 3 different replicas. Each replica will use the configured `size` for CPU and memory.
 
-### Pipeline Configuration Values
+The `size` parameter comprehend both CPU and memory. The value is a multiplier factor that is computed at runtime, starting from a base cpu/memory value.
+For example, if the default value for the memory - for a single replica - is 512 MB, specifying `size:  2` will make the pipeline to use 1024MB per-replica.
```

Comment:
where do I configure the values ?

can we add a section about it ?
we can write it here, or in a separate file and link it.

Reply:
```suggestion
For example, if the default value for the memory - for a single replica - is 512 MB, specifying `size: 2` will make the pipeline use 1024MB per-replica, as below:
```yaml
name: "Write to Cassandra"
topics:
    - name: "input-topic"
      creation-mode: create-if-not-exists
pipeline:
  - name: "Write to Cassandra"
    type: "sink"
    input: "input-topic"
    resources:
      size: 2
```
The default value for `parallelism` and `resources` is `1`. 
```